### PR TITLE
Fix exportmsg

### PR DESCRIPTION
**Pull request by ptersilie**
Fixes issues #164 and #159.

To be honest, the execution code is a bit of a mess, but I don't believe it's worth a rewrite given we will be moving to Eco2 soon. So I just tried to fix it without touching too much code.

### CODE REVIEW

**Comment by ltratt** [comment]
This looks fine to me. Can we get a bit more detail into the commit messages, and also rebase against master?

**Comment by ptersilie** [comment]
Updated the commits messages!

File: lib/eco/eco.py [hunk 1]
```diff
@@ -1627,17 +1627,21 @@ def __init__(self, window, parent):
...
-        else:
-            self.emit(self.signal_execute_fail)
+        try:
+            p = self.window.getEditor().export(run=True, source=str(ed.filename))
+            lang = self.window.getEditor().get_mainlanguage()
+            if p is False:
+                self.emit(self.signal_execute_fail, "program syntactically invalid")
+            else:
+                for line in iter(p.stdout.readline, b''):
+                    if lang == "PHP + Python":
```
Comment (ltratt) [inline lib/eco/eco.py]: Use `and` to join the two `if`s?

**Comment by ptersilie** [comment]
Addressed your comment. Let me know when I should squash!

**Comment by ltratt** [comment]
Works for me. Please squash.

**Comment by ptersilie** [comment]
Squashed!

**Review by ltratt — APPROVED** [review]

**Pull request merged**